Content.Server/_Goobstation/Devil/Contract/DevilContractSystem.cs

```diff
@@ -0,0 +1,506 @@
+            return;
+        }
+
+        // Check if devil is trying to sign first
+        if (args.Signer == contract.Comp.ContractOwner)
+        {
+            var tooEarlyPopup = Loc.GetString("devil-contract-early-sign-failed");
+            _popupSystem.PopupEntity(tooEarlyPopup, args.Signer, args.Signer, PopupType.MediumCaution);
+
+            args.Cancelled = true;
+        }
+
+    }
+
+    private void OnSignStep(Entity<DevilContractComponent> contract, ref SignSuccessfulEvent args)
+    {
+        // Determine signing phase
+        if (!contract.Comp.IsVictimSigned)
+            HandleVictimSign(contract, args.Paper, args.User);
+        else if (!contract.Comp.IsDevilSigned)
+            HandleDevilSign(contract, args.Paper, args.User);
+
+        // Final activation check
+        if (contract.Comp.IsContractFullySigned)
+            HandleBothPartiesSigned(contract);
+    }
+
+    private void HandleVictimSign(Entity<DevilContractComponent> contract, EntityUid signed, EntityUid signer)
+    {
+        contract.Comp.Signer = signer;
+        contract.Comp.IsVictimSigned = true;
+
+        if (TryComp<PaperComponent>(contract, out var paper))
+            paper.EditingDisabled = true;
+
+        _popupSystem.PopupEntity(Loc.GetString("contract-victim-signed"), signed, signer);
+    }
+
+    private void HandleDevilSign(Entity<DevilContractComponent> contract, EntityUid signed, EntityUid signer)
+    {
+        contract.Comp.IsDevilSigned = true;
+        _popupSystem.PopupEntity(Loc.GetString("contract-devil-signed"), signed, signer);
+
+        if (_mind.TryGetMind(signer, out var mindId, out var mind) &&
+            _mind.TryGetObjectiveComp<MeetContractWeightConditionComponent>(mindId, out var objectiveComp, mind))
+            objectiveComp.ContractWeight += contract.Comp.ContractWeight;
+    }
+
+    private void HandleBothPartiesSigned(Entity<DevilContractComponent> contract)
+    {
+        UpdateContractWeight(contract);
+        DoContractEffects(contract);
+    }
+
+    #endregion
+
+    #region Helper Events
+
+    public bool IsUserValid(EntityUid user, out string failReason)
+    {
+        if (HasComp<CondemnedComponent>(user)
+            || HasComp<SiliconComponent>(user)
+            || HasComp<BorgChassisComponent>(user))
+        {
+            failReason = Loc.GetString("devil-contract-no-soul-sign-failed");
+            return false;
+        }
+
+        if (HasComp<MindShieldComponent>(user)
+            && !HasComp<DevilComponent>(user))
+        {
+            failReason = Loc.GetString("devil-contract-mind-shielded-failed");
+            return false;
+        }
+
+        if (HasComp<PossessedComponent>(user))
+        {
+            failReason = Loc.GetString("devil-contract-early-sign-failed");
+            return false;
+        }
+
+        failReason = string.Empty;
+        return true;
+    }
+    public bool TryTransferSouls(EntityUid devil, EntityUid contractee, int added)
```

Comment:
Blank line between methods
```suggestion
    }

    public bool TryTransferSouls(EntityUid devil, EntityUid contractee, int added)
```